### DISCUSSION TITLE
Feature to save solution file upon receiving a UNIX signal

### DISCUSSION THREAD

**Discussion by TRPrasanna** [loganoz/horses3d#216] [Ideas]
It would be nice to have HORSES3D save solution files at any given time upon receiving a UNIX signal, say something like
```
kill -USR1 horses3d
```
Checkpoint signaling like this has been discussed [here](https://hpc-unibe-ch.github.io/slurm/checkpointing.html). This would come in handy when we have run our simulations for a long time but we accidentally (or intentionally, to save storage space) chose a long autosave interval.

**Comment by loganoz** [comment]
This is a really good suggestion. Do you have an idea of how to implement it? May be you can create a new branch with this functionality.

**Reply by TRPrasanna** [reply]
Actually, Fortran doesn't have native signal checkpointing features, but it seems GNU and Intel compilers provide extensions (beyond FORTRAN standard) that we can utilize. Another option is using an interface to a C program. These methods have been discussed [here](https://cyber.dabamos.de/programming/modernfortran/inter-process-communication.html). I can try implementing one of these.

**Reply by loganoz** [reply]
If you have the time, please try to implement it and keep us posted. Thanks!
